compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -17,15 +17,23 @@ pub struct Lexer<'a> {
     done: bool,
     skip_comments: bool,
     skip_whitespaces: bool,
+    src_id: SrcId,
 }
 
 pub type SpannedTokenResult = Result<SpannedToken, LexerErrorKind>;
 
 impl<'a> Lexer<'a> {
+    /// Given a source buffer of noir code, return all the tokens in the buffer
+    /// in order, along with any lexing errors that occurred.
+    pub fn lex_source(source: &'a str) -> (Tokens, Vec<LexerErrorKind>) {
+        let src_id = SrcId::default();
```

Comment:
I'm guessing this should be a `#[cfg(test)]` function as well.

Also, since we don't pass a `SrcId` to this function I think `lex_no_source` or `lex_raw_string` would be better names.

Reply:
removed